[vcpkg-tool-meson] fix pkgconfig data file installation on FreeBSD

SchaichAlonso opened the pull request:
Fixes #35002 .

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~SHA512s are updated for each updated download~
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

SchaichAlonso committed:
[fontconfig] fix pkgconfig file installation (microsoft/vcpkg#35002)

move pkgconfig script into the path indicated by the diagnositic emitted by
vcpkg (see comment 1 of microsoft/vcpkg#35002).

SchaichAlonso committed:
[fontconfig] bump portrevision (microsoft/vcpkg#35002)

SchaichAlonso committed:
[fontconfig] regenerate versions (microsoft/vcpkg#35002)

SchaichAlonso commented:
Seems the trivial attempt to fix it breaks more then it fixes...

dg0yt commented:
... there is nothing to fix on the supported platforms. The problem might be freebsd-only.

SchaichAlonso commented:
> ... there is nothing to fix on the supported platforms. The problem might be freebsd-only.

Indeed. It turns out meson has hardcoded the pkgconfig installation paths the different platforms, with [FreeBSD installing to libdata](https://github.com/mesonbuild/meson/blob/bb91cea0d66d8d036063dedec1f194d663399cdf/mesonbuild/modules/pkgconfig.py#L587) unless `install_dir` is specified [...]

SchaichAlonso committed:
[fontconfig] force installation of pc files (microsoft/vcpkg#35002)

Undo the last commit and patch meson to force the pc file installation paths
demanded by `vcpkg_fixup_pkgconfig`

SchaichAlonso commented:
This issue also seems to affect all meson ports, rather then only fontconfig -> cairo.

Rather then patching 20-ish ports to set `install_dir` explicitly, patching `vcpkg-tool-meson` to not provide different pc-file paths for different platforms might be more appropriate.

Neumann-A commented:
>  patching vcpkg-tool-meson

You mean directly patching meson. Im fine with patching meson. It is a hard coding mess any way. Installation directories/layout should be user specifiable but that is something meson will never understand

SchaichAlonso committed:
Revert "[fontconfig] regenerate versions (microsoft/vcpkg#35002)"

This reverts commit 8daf0e15a67c376779631442ce926236ba2ed2e4.

SchaichAlonso committed:
Revert "[fontconfig] bump portrevision (microsoft/vcpkg#35002)"

This reverts commit a9727bc32325238565428e3497ae564d2d255b86.

SchaichAlonso committed:
Revert "[fontconfig] force installation of pc files (microsoft/vcpkg#35002)"

This reverts commit c0c6e017c7ab255e5532428e68f9d7c6e1ae8aad.

SchaichAlonso committed:
Revert "[fontconfig] fix pkgconfig file installation (microsoft/vcpkg#35002)"

This reverts commit 616d4ce83e091cd128f4d896fcff81a1de4ed1b9.

SchaichAlonso committed:
[vcpkg-tool-meson] fix pc-file installation paths on FreeBSD (microsoft#35002)

SchaichAlonso committed:
[vcpkg-tool-meson] bump portversion (microsoft/vcpkg#35002)

SchaichAlonso committed:
[vcpkg-tool-meson] regenerate versions (microsoft/vcpkg#35002)

jimwang118 commented:
The function vcpkg_apply_patches is deprecated. Please use the PATCHES arguments to the "extract" helpers (for example, vcpkg_from_github() (https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_from_github))

jimwang118 commented:
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

SchaichAlonso committed:
[vcpkg-tool-meson] Use vcpkg_from_github to obtain source (microsoft/vcpkg#35002)

SchaichAlonso committed:
[vcpkg-tool-meson] Regenerate versions (microsoft/vcpkg#35002)

SchaichAlonso commented:
Rewrote the download code of the portfile and regenerated the patches.

Sorry it took me so long, I had connectivity issues last week.

jimwang118 commented on ports/vcpkg-tool-meson/vcpkg.json:
```diff
@@ -1,6 +1,7 @@
 {
   "name": "vcpkg-tool-meson",
   "version": "0.63",
```
Check that the latest meson version upstream is 1.3.0. Can't this version be updated to the latest one?

dg0yt replied:
The update to meson 1.3.0 is out of scope for this PR. This PR should be kept minimal to minimize merge avoid conflicts.
Cf. https://github.com/microsoft/vcpkg/pull/28084, https://github.com/microsoft/vcpkg/pull/35348.

jimwang118 commented on ports/vcpkg-tool-meson/portfile.cmake:
```diff
@@ -32,26 +28,22 @@ vcpkg_find_acquire_program(PYTHON3)
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO mesonbuild/meson
+    REF bb91cea0d66d8d036063dedec1f194d663399cdf
```
```suggestion
    REF "${VERSION}"
```

SchaichAlonso replied:
I can bump the version to ${VERSION}, which is `0.63`, however that will represent a major change, since despite being registered to vcpkg as `vcpkg-tool-meson-0.63#1`, bb91cea0d66d8d036063dedec1f194d663399cdf@mesonbuild/meson is actually meson `0.62.1` as the ref bit was forgotten to be updated in ce9f50f7aca5efa734b73f89434cbc1b44d8759e .

SchaichAlonso replied:
updated it. pango and its' dependencies seem to build with 0.63 over here, didn't check the others

SchaichAlonso replied:
The CI failed to build gobject-introspection with meson 0.63 so I'm reverting the meson version upgrade commit.

jimwang118 commented:
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

SchaichAlonso committed:
[vcpkg-tool-meson] Use adverticed version of meson (microsoft/vcpkg#35002)

vcpkg@ce9f50f7 has bumped vcpkg-tool-meson's adverticed version to 0.63, but
not updated the `ref` variable, which still points to 0.62.1.

Update meson to 0.63 and use ${VERSION} to avoid similar issues in the future,
and regenerate patches/hashes accordingly.

SchaichAlonso committed:
[vcpkg-tool-meson] regenerate versions (microsoft/vcpkg#35002)

SchaichAlonso commented:
Seems `gobject-instrospection` fails to build on x64-windows with meson-0.63, so i'll revert the meson version update part, as fixing meson version update regressions should be handled by their own tickets / PRs.

SchaichAlonso committed:
Revert "[vcpkg-tool-meson] regenerate versions (microsoft/vcpkg#35002)"

This reverts commit e9a424ef16d26e791e3e108e738982138084d41a.

SchaichAlonso committed:
Revert "[vcpkg-tool-meson] Use adverticed version of meson (microsoft/vcpkg#35002)"

Meson 0.63 can't build gobject-introspection on a tier 1 supported target.

Upgrading meson to 0.63 and handling regressions of that update is out of
scope of #35002 anyways, and should be handled either on it's own PR,
microsoft/vcpkg#28084 or microsoft/vcpkg#35348 .

BillyONeal commented:
LGTM
